runestone/assess/js/mchoice.js

```diff
@@ -294,6 +294,11 @@ MultipleChoice.prototype.restoreAnswers = function (data) {
             }
         }
     }
+    if (this.multipleanswers) {
```

Comment:
What answers were not getting properly restored?  I want to make sure I understand what you are trying to fix here...